find-closest-facility-from-point/src/main/java/com/esri/arcgismaps/sample/findclosestfacilityfrompoint/components/MapViewModel.kt

```diff
@@ -0,0 +1,199 @@
+        Facility(Point(-1.3042193400557665E7, 3862448.873041752, SpatialReference.webMercator())),
+        Facility(Point(-1.3046882875518233E7, 3862704.9896770366, SpatialReference.webMercator())),
+        Facility(Point(-1.3040539754780494E7, 3862924.5938606677, SpatialReference.webMercator())),
+        Facility(Point(-1.3042571225655518E7, 3858981.773018156, SpatialReference.webMercator())),
+        Facility(Point(-1.3039784633928463E7, 3856692.5980474586, SpatialReference.webMercator())),
+        Facility(Point(-1.3049023883956768E7, 3861993.789732541, SpatialReference.webMercator()))
+    )
+
+    // Create graphic overlay of facilities to display on the map
+    private val facilityGraphicsOverlay = GraphicsOverlay()
+
+    // Create graphic overlay of incident to display on map tapped
+    private val incidentGraphicsOverlay = GraphicsOverlay()
+
+    // Create a ViewModel to handle dialog interactions
+    private val messageDialogVM: MessageDialogViewModel = MessageDialogViewModel()
+
+    // Add the graphics overlays to be used by the composable MapView
+    val graphicsOverlays = listOf(facilityGraphicsOverlay, incidentGraphicsOverlay)
+
+    // Create a streets basemap layer and change the position of map to center around San Diego
+    val map = ArcGISMap(BasemapStyle.ArcGISStreets).apply {
+        initialViewpoint = Viewpoint(
+            latitude = 32.727,
+            longitude = -117.1750,
+            scale = 100000.0
+        )
+    }
+
+    init {
+        // Create the facility symbol
+        val facilityUrl = application.getString(R.string.hospital_symbol_url)
+        val facilitySymbol = PictureMarkerSymbol(facilityUrl).apply {
+            height = 30F
+            width = 30F
+        }
+
+        // Create a graphic for each facility and add them to the graphics overlay
+        for (facility in facilitiesList) {
+            val facilityGraphic = Graphic(
+                geometry = facility.geometry,
+                symbol = facilitySymbol
+            )
+            facilityGraphicsOverlay.graphics.add(facilityGraphic)
+        }
+    }
+
+    /**
+     * Retrieve the tapped point [event], update the incident, and initiate the process
+     * to find the route between the incident and closest facility.
+     */
+    fun onSingleTapConfirmed(event: SingleTapConfirmedEvent) {
+
+        // Retrieve the tapped map point from the SingleTapConfirmedEvent
+        val mapPoint: Point = event.mapPoint ?: return messageDialogVM.showMessageDialog(
+            title = "No map point retrieved from tap.",
+            description = ""
+        )
+
+        // Create an incident on the tapped point
+        val incidentPoint = Incident(mapPoint)
+
+        // Clear and update the incident graphic on the map
+        updateIncidentGraphicPosition(incidentPoint)
+
+        // Find the closest facility to the incident
+        viewModelScope.launch {
+            findRoute(Incident(mapPoint))
```

Comment:
You have already created a `val` above on line 112. 
```suggestion
            findRoute(incidentPoint)
```